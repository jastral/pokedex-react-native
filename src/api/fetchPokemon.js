import axios from "axios";

export async function fetchPokemon(pokeId) {
  const POKE_URL = `https://pokeapi.co/api/v2/pokemon/${pokeId}`;

  const response = await axios
    .get(POKE_URL)
    .catch((error) => console.error(`Error: ${error}`));

  return response;
}
