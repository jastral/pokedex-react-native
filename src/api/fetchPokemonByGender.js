import axios from "axios";

export async function fetchPokemonByGender(pokeId) {

  const POKE_GENDER_URL = `https://pokeapi.co/api/v2/pokemon-species/${pokeId}/`;

  const response = await axios
    .get(POKE_GENDER_URL)
    .then((data) => {
      const genderRate = data?.data?.gender_rate; // Gender rate is a number between 0 and 8
      let gender = "Genderless"; // Default value if the Pokemon has a gender rate of -1 or 0
      if (genderRate > 0) {
        gender = genderRate % 2 === 0 ? "Female" : "Male"; // Calculate gender based on gender rate
      }
      return gender;
    })
    .catch((error) => console.error(`Error: ${error}`));

  return response;
}

