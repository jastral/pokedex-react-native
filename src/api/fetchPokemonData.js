import axios from "axios";
import { fetchPokemon } from "./fetchPokemon";

export async function fetchPokemonData(pokeId) {
  const POKE_SPECIES_URL = `https://pokeapi.co/api/v2/pokemon-species/${pokeId}`;
  const POKE_TYPE_URL = `https://pokeapi.co/api/v2/type/${pokeId}`;

  const pokemonData = await fetchPokemon(pokeId);

  const pokemonSpeciesData = await axios
    .get(POKE_SPECIES_URL)
    .catch((error) => console.error(`Error: ${error}`));

  const pokemonTypeData = await axios
    .get(POKE_TYPE_URL)
    .catch((error) => console.error(`Error: ${error}`));

  const pokemon = pokemonData?.data;
  const pokemonSpecies = pokemonSpeciesData?.data;
  const pokemonType = pokemonTypeData?.data;


  return {pokemon, pokemonSpecies, pokemonType};
}
