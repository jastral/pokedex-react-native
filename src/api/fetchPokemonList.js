import axios from "axios";
import { fetchPokemon } from "./fetchPokemon";

// export const POKE_LIST_URL = "https://pokeapi.co/api/v2/pokemon";



export async function fetchPokemonList() {

  // const offset = 20 * (page - 1);
  const POKE_LIST_URL = `https://pokeapi.co/api/v2/pokemon?offset=0&limit=786`;

  const res = await axios
    .get(POKE_LIST_URL)
    .catch((error) => console.error(`Error: ${error}`));

  const data = res?.data?.results;

  const promises = data.map(async (pokemon) => {
    const { url } = pokemon;
    const pokeId = url.substring(34, url.length - 1);
    const response = await fetchPokemon(pokeId);
    const type = response?.data?.types;

    return {...pokemon, pokeId, type}
  })
  const pokemons = Promise.all(promises);



  return pokemons;
}
