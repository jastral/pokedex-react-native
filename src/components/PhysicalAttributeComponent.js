import { useEffect, useState } from "react";
import { View, Text, StyleSheet, Pressable } from "react-native";
import { fetchPokemonByGender } from "../api/fetchPokemonByGender";

export default function PhysicalAttributeComponent(pokemonData) {
  const pokemon = pokemonData?.data.pokemon;
  const pokemonSpecies = pokemonData?.data.pokemonSpecies;
  const pokemonType = pokemonData?.data.pokemonType;

  const [gender, setGender] = useState('');

  const id = pokemon?.id; 

  async function getPokemonGender() {
    const data = await fetchPokemonByGender(id);

    return setGender(data);
  }

  useEffect(() => {
    getPokemonGender();
  });

  const EggGroups = pokemonSpecies?.egg_groups.map((eggGroup, idx) => {
    return (
      <Text style={styles.text} key={idx}>
        {eggGroup?.name}
      </Text>
    );
  });

  const Abilities = pokemon?.abilities.map((ability, idx) => {
    return (
      <Text style={styles.text} key={idx}>
        {ability?.ability?.name}
      </Text>
    );
  });

  const Types = pokemon?.types.map((type, idx) => {
    return (
      <Text style={styles.text} key={idx}>
        {type?.type?.name}
      </Text>
    );
  });

  const WeakAgainst = pokemonType?.damage_relations?.double_damage_from.map(
    (damage, idx) => {
      return (
        <Text style={styles.text} key={idx}>
          {damage?.name}
        </Text>
      );
    }
  );

  return (
    <View style={styles.pAContainer}>
      <Text style={styles.heading}>Height:</Text>
      <Text style={styles.text}>{pokemon?.height}</Text>
      <Text style={styles.heading}>Weight:</Text>
      <Text style={styles.text}>{pokemon?.weight}</Text>
      <Text style={styles.heading}>Gender:</Text>
      <Text>{gender}</Text>
      <Text style={styles.heading}>Egg Group:</Text>
      <View>{EggGroups}</View>
      <Text style={styles.heading}>Abilities:</Text>
      <View>{Abilities}</View>
      <Text style={styles.heading}>Types:</Text>
      <View>{Types}</View>
      <Text style={styles.heading}>Weak Against:</Text>
      <View>{WeakAgainst}</View>
    </View>
  );
}

const styles = StyleSheet.create({
  heading: {
    fontWeight: "bold",
    fontSize: 14,
  },
  text: {
    fontSize: 14,
  },
  pAContainer: {
    
  }
});
