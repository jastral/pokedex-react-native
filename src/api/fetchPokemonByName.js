import axios from "axios";
import { fetchPokemonList } from "./fetchPokemonList";

export async function fetchPokemonByName(pokeName) {
  const pokemonList = await fetchPokemonList();

  const filteredList = pokemonList.filter((pokemon) => pokemon?.name.includes(pokeName));
  console.log(filteredList);

  return filteredList;
}
