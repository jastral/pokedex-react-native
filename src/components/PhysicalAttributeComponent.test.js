import React from 'react';
import {ActivityIndicator, FlatList, Text, TextInput} from 'react-native';
import renderer from 'react-test-renderer';
import PhysicalAttributeComponent from './PhysicalAttributeComponent';
import { pokemonData } from '../mock/data';
import { fetchPokemonByGender } from '../api/fetchPokemonByGender';

it(`Checking Carousel component Rendering`, () => {
  const tree = renderer.create(<PhysicalAttributeComponent data={pokemonData}></PhysicalAttributeComponent>);
  expect(tree).toMatchSnapshot();
});

