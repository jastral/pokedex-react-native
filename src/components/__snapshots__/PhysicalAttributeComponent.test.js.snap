// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checking Carousel component Rendering 1`] = `
<View>
  <Text
    style={
      {
        "fontSize": 14,
        "fontWeight": "bold",
      }
    }
  >
    Height:
  </Text>
  <Text
    style={
      {
        "fontSize": 14,
      }
    }
  >
    7
  </Text>
  <Text
    style={
      {
        "fontSize": 14,
        "fontWeight": "bold",
      }
    }
  >
    Weight:
  </Text>
  <Text
    style={
      {
        "fontSize": 14,
      }
    }
  >
    69
  </Text>
  <Text
    style={
      {
        "fontSize": 14,
        "fontWeight": "bold",
      }
    }
  >
    Gender:
  </Text>
  <Text />
  <Text
    style={
      {
        "fontSize": 14,
        "fontWeight": "bold",
      }
    }
  >
    Egg Group:
  </Text>
  <View>
    <Text
      style={
        {
          "fontSize": 14,
        }
      }
    >
      monster
    </Text>
    <Text
      style={
        {
          "fontSize": 14,
        }
      }
    >
      plant
    </Text>
  </View>
  <Text
    style={
      {
        "fontSize": 14,
        "fontWeight": "bold",
      }
    }
  >
    Abilities:
  </Text>
  <View>
    <Text
      style={
        {
          "fontSize": 14,
        }
      }
    >
      overgrow
    </Text>
    <Text
      style={
        {
          "fontSize": 14,
        }
      }
    >
      chlorophyll
    </Text>
  </View>
  <Text
    style={
      {
        "fontSize": 14,
        "fontWeight": "bold",
      }
    }
  >
    Types:
  </Text>
  <View>
    <Text
      style={
        {
          "fontSize": 14,
        }
      }
    >
      grass
    </Text>
    <Text
      style={
        {
          "fontSize": 14,
        }
      }
    >
      poison
    </Text>
  </View>
  <Text
    style={
      {
        "fontSize": 14,
        "fontWeight": "bold",
      }
    }
  >
    Weak Against:
  </Text>
  <View>
    <Text
      style={
        {
          "fontSize": 14,
        }
      }
    >
      flying
    </Text>
    <Text
      style={
        {
          "fontSize": 14,
        }
      }
    >
      psychic
    </Text>
    <Text
      style={
        {
          "fontSize": 14,
        }
      }
    >
      fairy
    </Text>
  </View>
</View>
`;
